Documentation/README.md

```diff
@@ -0,0 +1,745 @@
+communicate with this one. The examples that follow assume a Linux operating 
+system, but most of these commands will work cross-platform.
+
+Let's create the CA: 
+
+`cd /etc/nebula ; /opt/nebula/nebula-cert ca -name my_certificate_authority` 
+
+When you run this command, it will create two files - `/etc/nebula/ca.key` 
+and `/etc/nebula/ca.crt`. These are your 
+[public and private keys](https://en.wikipedia.org/wiki/Public-key_cryptography) 
+for the Certificate Authority.
+
+**YOU MUST KEEP THE FILE `ca.key` PROTECTED**
+
+### Client Certificates
+At this point, you need to have a bit of an idea on just how your network 
+should look! Do you want every trusted host to reach every trusted host? 
+Do you only want "management" machines to be able to reach "servers"? Do you 
+only want your "web servers" to be able to reach your "database servers" 
+or... something more?! This will define how you build your next few steps.
+
+Most people are going to want to segregate things a little bit, so this next 
+block will be based on a model where your "client" machines are in a group 
+called "client" and your servers are in a group called "servers". "Clients" 
+can talk to "servers" on any port, but "servers" can't initiate connections 
+to "clients".
+
+So, let's build our first node certificate. The simplest way to do this is on 
+the CA server, like this: 
+
+`cd /etc/nebula ; /opt/nebula/nebula-cert sign -name client1 -ip 192.0.2.1/24 -groups client`
+
+Fab. 
+
+This creates a `client1.key` and `client1.crt` file, and then uses the `ca.key` 
+to sign the `client1.crt` file. This key pair, much like the CA, is valid for 
+1 year. There will be some notes later on how to do things to mitigate against 
+issues with these certificates expiring.
+
+You need to transfer the `client1.*` and `ca.crt` file to your client. 
+
+Now let's build your second node certificate - this is for your server. 
+
+`cd /etc/nebula ; /opt/nebula/nebula-cert sign -name server1 -ip 192.0.2.128/24 -groups server`
+
+Great! We've got certificates to work between nodes! Now, let's build your 
+configuration!
+
+### Client configuration
+There's an [example configuration file](/examples/config.yml) in the repository
+but this is going to expand out some details on each block to help you decide
+if you want to use these features!
+
+Let's start at the top!
+
+#### PKI
+
+```yaml
+pki:
+  ca: /etc/nebula/ca.crt
+  cert: /etc/nebula/host.crt
+  key: /etc/nebula/host.key
+  #blacklist:
+  # - c99d4e650533b92061b09918e838a5a0a6aaee21eed1d12fd937682865936c72
+```
+
+This block here tells your file where it can find your PKI files.
+
+There's also a commented out section called "blacklist". A blacklist is a
+collection of certificates we MUST NOT talk to - typically compromised or lost
+machines. Those machines must not be able to talk to the network, and so should
+be added as a list of "fingerprints" to ignore.
+
+<!-- TODO: Document how to get the fingerprints -->
```

Comment:
To retrieve the thumbprint, simply use 

`./nebula-cert print -path /path/to/your/cert.crt | grep Fingerprint | awk {'print $2'}`

Reply:
Added (practically) verbatim in 8bc14b0